keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -369,6 +371,43 @@ of the following scenarios:
 2. A new Node results in a Service that is able to achieve an endpoint
    distribution below 20% for the first time.
 
+### Manual enforcement of Topology Aware Routing by zone
+
+The algorithm proposed for mitigating the risk of overload works best with large
+numbers of endpoints and nodes, such that the percentage thresholds are unlikely
+to be broached with service restarts and node additions/removals within zones.
+There are a number of use cases however where the proposed algorithm for automatic
+assignments of zone hints is not suitable:
+
+1. Small multi-zone clusters with a low number of zones and endpoints. For 
+example a rolling restart with a service in a 3 zone cluster with 2-2-2 
+endpoint distribution will cause the proposed algorithm to remove hints for the 
+service during the restart, this could have severe impact for high latency 
+environments.
+2. Multi-zone clusters that deploy applications in a subset of zones. Unless 
+all zones in a cluster contain an endpoint then Topology Hints will not be 
+applied.
+
+As such we should ensure that service owners have a way to enforce traffic is
+sent to endpoints in the same zone, regardless of node/endpoint distribution.
+If there is no endpoint in the zone, then the default service load balancing
+strategy should be used on all endpoints for the service regardless of topology.
+
+### Manual enforcement of Topology Aware Routing by multiple metrics
+
+The above proposal also ties in nicely to KEP: 3015. As such we could combine both 
+zonal and node based topology routing preferences into a new annotation 
+`topology-aware-routing` with the values option of `[node, zone, region, auto]`.
+
+The first 3 values would make routing decisions preferring the given metric, for
+example `topology-aware-routing=zone` would implement the previously described
+behaviour. `topology-aware-routing=node` would send traffic to the local node if
+an endpoint exists on it as described in KEP 3015.
+
+The `auto` option would combine these metrics such that it would prefer endpoints
+by node, if none present then by zone, and if none present by zone then by region, 
+finally falling back to the default behaviour if there are no endpoints.
```

Comment:
this does not solve the problem of overloading endpoints on the same zone, see #overload section

Reply:
Yeah I know, the entire point of this PR though is that by protecting me as a cluster admin from overload (a valid problem) but not giving me any tools/options to override that protection, it hurts me far more in terms of service instability.

I understand the overload issue, and it's something that's good to have options to help avoid, but I'm proposing giving cluster admins the ability to also say "always force topological routing if there is an instance available, and let me ensure my services are sufficiently scaled in the appropriate regions to avoid overload" - this should be documented and highlighted as an issue, but the lack of options in this space is frustrating.

Feels like we should just be giving multiple options rather than enforcing a kind of "we know better than you" option as the only possibility.